portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java

```diff
@@ -37,29 +39,34 @@
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         long id = Long.parseLong(request.getParameter("id"));
 
-        //get comment entity corresponding to id from datastore
-        Key commentEntityKey = KeyFactory.createKey("Comment", id);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Transaction txn = datastore.beginTransaction();
 
         try {
+            //get comment entity corresponding to id from datastore
+            Key commentEntityKey = KeyFactory.createKey("Comment", id);
             Entity commentEntity = datastore.get(commentEntityKey);
-
+            
             //calculate the number of likes using the previous number of likes
             long numLikes = (long) commentEntity.getProperty("numLikes") + 1;
```

Comment:
I noticed you are storing all the likes in the database. To eliminate some concurrency/sync issues, you could just do a database query for the number of entities in the like database that match the user ID and comment ID instead of storing the actual number as a separate entity.

Reply:
I'm only storing the likes if the user is logged in. Logged out users can also view and like comments, but don't have an associated user ID.

Reply:
Ah I didn't catch that. In that case, is there anything stopping a logged out user from opening a bunch of tabs and liking a comment multiple times?

I'd suggest that users should be logged in to like/unlike a comment (similar to how reddit handles this)

Reply:
That's true. I'll work on that right now!

Reply:
Alright, I deleted the numLikes property and queried into datastore to count the number of Like entities for each comment. This also eliminated the issue of the `IllegalArgumentException` because I no longer have to deal with two different entities.